lib/ansible/modules/network/vyos/vyos_config.py

```diff
@@ -223,7 +223,7 @@ def run(module, result):
         result['changed'] = True
 
     if module._diff:
-        result['diff'] = diff
+        result['diff'] = {'prepared': diff}
```

Comment:
Is there a specific reason for moving generated diff under prepared key?

Reply:
@ganeshrn Thank you for comment.

Just for printing diff.
I think printing diff need (`before`, `after`) or `prepared` keys. 
https://github.com/ansible/ansible/blob/9c5d40ff152a8a94dc226735ce43593ebbb10104/lib/ansible/plugins/callback/__init__.py#L151-L219

Is it wrong?

Reply:
I get it now, so adding diff as part of `prepared` key results is rendering diff in a readable format in output logs.
The only concern I have here is this change might break backward compatibility for playbook that depend on `diff` key

Reply:
Ah, that may be the problem.
Do you have any idea to keep backward compatibility?

Reply:
Note that this line was only added recently (https://github.com/ansible/ansible/pull/41846/files) for Ansible 1.7 development branch, i.e. it has not been added to a release.
So your change should be safe.

Reply:
In fact, @ganeshrn, you added it a couple of weeks ago ;)

Reply:
@felixfontein Good catch. Thanks for pointing out :)
@hitsumabushi diff is added as part of result for most network platforms, somehow is was missed in vyos_config for earlier release. I added `diff` in vyos_config during the recent cliconf refactor work. Looking at the PR now assumed that it was present since ealier releases.